onnxruntime/core/graph/graph.cc

```diff
@@ -4175,6 +4175,15 @@ Status Graph::AddExternalInitializersToGraphProtoImpl(
       size_t tensor_bytes_size = raw_data.size();
       if (tensor_bytes_size < model_saving_options.initializer_size_threshold) {
         *output_proto = initializer;
+        // Data with size above the threshold is written into the new external initializer file
+        // Data with size below the threshold should be kept inside the new model file
+        // instead of leaving it in the old external initializer file for the old Onnx file
+        if (initializer.data_location() == TensorProto_DataLocation_EXTERNAL) {
+          TensorShape shape(initializer.dims());
+          auto data_size = shape.Size();
+          utils::PackDataIntoTensorProto(*output_proto, raw_data, data_size);
```

Comment:
I think it would also be valid to store the data in TensorProto's raw_data field. Could something like the following work and avoid need to make a new util function?

```C++
output_proto->set_raw_data(std::move(std::string(raw_data.begin(), raw_data.end())));
```